numcodecs/tests/common.py

```diff
@@ -147,6 +147,10 @@ def check_encode_decode_array(arr, codec):
     codec.decode(enc, out=out)
     assert_array_items_equal(arr, out)
 
+    enc = codec.encode(arr)
+    dec = codec.decode(ensure_ndarray(enc))
+    assert_array_items_equal(arr, dec)
```

Comment:
Seems this worked for everything except `Pickle` on Python 2 as demonstrated by [this CI failure]( https://travis-ci.org/zarr-developers/zarr/jobs/461902060#L2893 ). This is likely a consequence of the improved buffer handling under the hood, which `MsgPack`, `JSON`, etc. all use already. `Pickle` was the only one not using these functions, which this PR fixes. So testing this generally seems to be fine. All the other codecs already test decoding with `ndarray`.